### DISCUSSION TITLE
Localization Available Languages

### DISCUSSION THREAD

**Discussion by arunswarnam** [eclipse-theia/theia#11307] [Q&A] [answered]
Hi! 
 I have a question related to this [line](https://github.com/eclipse-theia/theia/blob/master/packages/core/src/node/i18n/localization-provider.ts#L53) . I see Theia allows a developer to bring in their own localization via [registerLocalizationFromRequire()](https://theia-ide.org/docs/i18n/#localizing-your-extension). 
1.  I am trying to understand why are we filtering out the languageId's that only came from a language pack? I see there is an `all` flag but that is not currently set in any calls to `getAvailableLanguages(..)` in the Theia codebase.

I my case I think I have an option to customize either the `LocalizationBackendContribution` (caller)  or the `LocalizationProvider` (callee) but I am trying to see if I am missing to see an underlying reason why the `all` flag  wasn't enabled by default.

**Comment by msujew** [comment, marked as answer]
Hi @arunswarnam, while Theia contains some localizations for itself (see [this directory](https://github.com/eclipse-theia/theia/tree/master/packages/core/i18n)), most of the localizations are actually contributed by language packs. If we were to enable the `all` flag, 90% of the app would remain untranslated, since we depend on vscode language packs to reduce the maintenance effort for translations.

That's the reason we only show languages which actually have a language pack installed.

**Reply by arunswarnam** [reply]
Makes sense. Thank you!
